integration/spark/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala

```diff
@@ -316,4 +324,17 @@ object IndexServer extends ServerInterface {
       Array(new Service("security.indexserver.protocol.acl", classOf[ServerInterface]))
     }
   }
+
+  def startAgingFolders(): Unit = {
+    val runnable = new Runnable() {
+      def run() {
+        val age = System.currentTimeMillis() - agePeriod.toLong
+        CarbonUtil.agingTempFolderForIndexServer(age)
+        LOGGER.info(s"Complete age temp folder ${CarbonUtil.getIndexServerTempPath}")
+      }
+    }
+    val ags: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor
+    ags.scheduleAtFixedRate(runnable, 1000, 3600000, TimeUnit.MICROSECONDS)
```

Comment:
Why are you scheduling this after 3s.. I think this scheduling rate is too fast. 

Better to execute this thread at a fixed delay of 1 hour.

Reply:
delay 1 hour may be to late, if something wrong happens, 1hour is enough to cause a relatively large impcact。
may be we can delay for 5 min, what do you think?

Reply:
the rate is 3 hours, about the delay time, i thinks it is ok, delay 1s or delay 5min or delay 1hour the effect is almost the same.
The test cases are covered here. If there is too much delay, the execution of test cases will be affected.
so kunal is there no need to modify the delay here?

Reply:
the unit you used is microseconds not milliseconds, better to keep it as 5 min. 3s very less duration

Reply:
oh sorry will change to milliseconds, how about keep it as 3hours.

Reply:
done